std/src/main/resources/std/list.mgn

```diff
@@ -61,3 +61,22 @@ def take := Y (\rec.n.xs. (isnil xs) [] ((gt n 0) (cons (head xs) (rec (pred n)
 def takeWhile := Y (\rec.p.xs. (isnil xs) [] ((p (head xs)) (cons (head xs) (rec p (tail xs))) []))
 
 def dropWhile := Y (\rec.p.xs. (isnil xs) [] ((p (head xs)) (rec p (tail xs)) xs))
+
+/* Concatenation of lists `xs` and `ys`, by reconstruction of `xs`
+ * as a prefix of `ys`
+ */
+def concat := \xs.ys. Y (\rec.zs. (isnil zs)
+				   ys
+				   (cons (head zs)
+				   	 (rec (tail zs)))) xs
+
+/* Returns a tuple (a, b) where
+ * a is a boolean indicating if the nth value is present
+ * b is the nth element of the sequence or nil, if a is false
+ */
+def nth := Y (\rec.xs.n. (isnil xs)
```

Comment:
I see it more likely to be common to bind `n` first instead of `xs`. For instance, mapping over a list of `lines` may look like:

```morganey
λ> map (nth 3) lines    // take the third character of each line
```

So, I'd suggest to swap the arguments of this function.